src/Compilers/VisualBasic/Portable/Parser/Parser.vb

```diff
@@ -4186,7 +4186,11 @@ checkNullable:
                 ' TODO - GreenSepList knows its error count. Expose it instead of recomputing it.
                 If propertyParameters.Count = 0 Then
                     Dim unexpected = ResyncAt({SyntaxKind.AsKeyword, SyntaxKind.ImplementsKeyword, SyntaxKind.EqualsToken})
-                    closeParen = closeParen.AddTrailingSyntax(unexpected)
+                    If unexpected.Count > 0 Then
+                        closeParen = closeParen.AddTrailingSyntax(unexpected, ERRID.ERR_ExpectedEOS)
```

Comment:
Might be better as an error that indicates `As` was expected? @AlekseyTs, do you have thoughts?

Reply:
@333fred I kept the same behavior as the native compiler as it will/should be consistent with other similar cases.

Reply:
Alright, that's enough reason for me.